src/common/arrow-ffi/src/lib.rs

```diff
@@ -82,14 +117,98 @@ pub fn field_to_py(
     Ok(field.into())
 }
 
+fn convert_field(field: daft_arrow::datatypes::Field) -> arrow_schema::Field {
```

Comment:
Can we put this in `daft_arrow`, since it will probably be useful for the DataType migration anyways?